velox/type/tests/DecimalTest.cpp

```diff
@@ -193,6 +193,22 @@ TEST(DecimalTest, valueInPrecisionRange) {
       DecimalUtil::kLongDecimalMin - 1, LongDecimalType::kMaxPrecision));
 }
 
+TEST(DecimalTest, computeAverage) {
+  auto validateSameValues = [](int128_t value, int64_t count) {
+    SCOPED_TRACE(fmt::format("value={} count={}", value, count));
+    int128_t sum = 0;
+    int64_t overflow = 0;
+    for (int64_t i = 0; i < count; i++) {
+      overflow += DecimalUtil::addWithOverflow(sum, sum, value);
+    }
+    int128_t avg;
+    DecimalUtil::computeAverage(avg, sum, count, overflow);
+    ASSERT_EQ(avg, value);
+  };
+  validateSameValues(DecimalUtil::kLongDecimalMin, 9);
```

Comment:
IIRC, the average would converge for some values of `count`.
Would you please add tests with different counts.

Reply:
I agree. We should test with a larger count say
```
  int128_t sum = 0;
  int64_t overflow = 0;
  int64_t count = 0;
  for (int i = 0 ; i < 1'000'000; i++) {
    overflow +=
        DecimalUtil::addWithOverflow(sum, sum, DecimalUtil::kLongDecimalMax);
    count++;
    int128_t avg = 0;
    DecimalUtil::computeAverage(avg, sum, count, overflow);
    ASSERT_EQ(avg, DecimalUtil::kLongDecimalMax);
  }
```